### PR TITLE
add support for bareudp link type

### PR DESCRIPTION
**Pull request by bersoare**
this PR adds support for bareudp link attributes.

2 links with similar attributes - one created using this code, another one using iproute2:
```
13: bareudp1: <POINTOPOINT,MULTICAST,NOARP> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
    link/none  promiscuity 0 minmtu 68 maxmtu 65527
    bareudp dstport 6636 ethertype mpls_uc srcportmin 12345 multiproto addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535
14: foo99: <POINTOPOINT,MULTICAST,NOARP> mtu 1500 qdisc noop state DOWN mode DEFAULT group default
    link/none  promiscuity 0 minmtu 68 maxmtu 65527
    bareudp dstport 6635 ethertype mpls_uc srcportmin 12345 multiproto addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535
```

### CODE REVIEW

File: link.go [hunk 1]
```diff
@@ -1283,11 +1283,27 @@ func (ipoib *IPoIB) Type() string {
 	return "ipoib"
 }
 
+type Bareudp struct {
```
Comment (aboch) [inline link.go:1286]: Should _UDP_ be capitalized?
If I look at https://golang.org/pkg/net/ _TCP_ and _UDP_ are capitalized in many function/types names

Comment (bersoare) [reply link.go]: i have implemented the changes requested. you mind having a look again? 

thanks

**Comment by aboch** [comment]
LGTM

But I we need to wait for @vishvananda to see if he can revive Travis CI integration

**Comment by aboch** [comment]
Please rebase and force-push so that CI (now fixed)  is triggered

**Comment by aboch** [comment]
LGTM

**Pull request merged**